File: frontend/src/lib/projects.ts
import { authFetch } from './api'

export type Project = {
  id: number
  name: string
  description: string | null
  key: string
  lead_user_id: number | null
  status: string
  require_requirement_link: boolean
  auto_resolve_days: number | null
  ai_impact_suggestions: boolean
  default_severity: string
  created_at: string
}

export type ProjectUpdatePayload = {
  name: string
  description?: string
  key: string
  lead_user_id?: number | null
  status: string
  require_requirement_link: boolean
  auto_resolve_days?: number | null
  ai_impact_suggestions: boolean
  default_severity: string
}

export async function listProjects(): Promise<Project[]> {
  return authFetch<Project[]>('/projects')
}

export async function getProject(id: number): Promise<Project> {
  return authFetch<Project>(`/projects/${id}`)
}

export async function createProject(payload: {
  name: string
  description?: string
}): Promise<Project> {
  return authFetch<Project>('/projects', { method: 'POST', body: payload })
}

export async function updateProject(id: number, payload: ProjectUpdatePayload): Promise<Project> {
  return authFetch<Project>(`/projects/${id}`, { method: 'PUT', body: payload })
}

export async function deleteProject(id: number): Promise<void> {
  await authFetch<void>(`/projects/${id}`, { method: 'DELETE' })
}

File: backend/schemas/projects.py
from datetime import datetime

from pydantic import BaseModel


class ProjectCreate(BaseModel):
    name: str
    description: str | None = None


class ProjectUpdate(BaseModel):
    name: str
    description: str | None = None
    key: str
    lead_user_id: int | None = None
    modules: list[str] = []
    status: str
    require_requirement_link: bool
    auto_resolve_days: int | None = None
    ai_impact_suggestions: bool
    slack_alerts_enabled: bool
    retention_days: int
    default_severity: str


class ProjectResponse(BaseModel):
    id: int
    name: str
    description: str | None
    key: str
    lead_user_id: int | None
    modules: list[str]
    status: str
    require_requirement_link: bool
    auto_resolve_days: int | None
    ai_impact_suggestions: bool
    slack_alerts_enabled: bool
    retention_days: int
    default_severity: str
    created_at: datetime

    class Config:
        from_attributes = True
